crates/pallet-domains/src/staking.rs

```diff
@@ -0,0 +1,366 @@
+//! Staking for domains
+
+use crate::pallet::{
+    DomainStakingSummary, NextOperatorId, OperatorIdOwner, OperatorPools, PendingTransfers,
+};
+use crate::{BalanceOf, Config, FreezeIdentifier};
+use codec::{Decode, Encode};
+use frame_support::traits::fungible::{Inspect, InspectFreeze, MutateFreeze};
+use frame_support::traits::tokens::{Fortitude, Preservation};
+use frame_support::{ensure, PalletError};
+use scale_info::TypeInfo;
+use sp_core::Get;
+use sp_domains::{DomainId, EpochIndex, ExecutorPublicKey, OperatorId};
+use sp_runtime::traits::{CheckedAdd, Zero};
+use sp_runtime::Percent;
+use sp_std::vec::Vec;
+
+/// Type that represents an operator pool details.
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct OperatorPool<Balance, Share> {
+    pub signing_key: ExecutorPublicKey,
+    pub current_domain_id: DomainId,
+    pub minimum_nominator_stake: Balance,
+    pub nomination_tax: Percent,
+    /// Total active stake for the current pool.
+    pub current_total_stake: Balance,
+    /// Total stake for the current pool in the next epoch.
+    pub next_total_stake: Balance,
+    /// Total shares of the nominators and the operator in this pool.
+    pub total_shares: Share,
+    pub is_frozen: bool,
+}
+
+/// Type that represents a nominator's details under a specific operator pool
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct Nominator<Share> {
+    pub shares: Share,
+}
+
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub enum Transfer<Balance> {
+    Withdraw(Balance),
+    Deposit(Balance),
+}
+
+/// Type that represents a pending transfer
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct PendingTransfer<NominatorId, Balance> {
+    pub nominator_id: NominatorId,
+    pub transfer: Transfer<Balance>,
+}
+
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct StakingSummary<OperatorId, Balance> {
+    /// Current epoch index for the domain.
+    pub current_epoch_index: EpochIndex,
+    /// Total active stake for the current epoch.
+    pub current_total_stake: Balance,
+    /// Total stake for the next epoch.
+    pub next_total_stake: Balance,
```

Comment:
Can we use `current_total_domain_stake` and `next_total_domain_stake` here? StakingSummary and OperatorPool have the same fields `current_total_stake`/`next_total_stake`, which is really confusing to me.

Reply:
I have removed the `next_total_stake` completely in the withdrawls PR as this is not required.
As for the renaming, the field is under Domain's stake summary and adding `domain` to the field is redundant IMHO.
If you have a string opinion, happy to update 👍🏼

Reply:
In the code, likely okay as they are concise, but it's really hard to tell which it's without looking into the context in the spec, I believe the spec can become much more readable this way :P